templates/cluster/aws-standalone-cp/templates/k0scontrolplane.yaml

```diff
@@ -15,15 +15,32 @@ spec:
       - --enable-cloud-provider
       - --kubelet-extra-args="--cloud-provider=external"
       - --disable-components=konnectivity-server
+    {{- if and (.Values.k0s.auth) (.Values.k0s.files) }}
+    {{- if .Values.k0s.auth.enabled }}
     files:
-      {{- if .Values.k0s.auth.enabled }}
-      - content: |
-          {{- with .Values.k0s.auth.config }}
-          {{- toYaml . | nindent 14 }}
-          {{- end }}
-        permissions: "0644"
-        path: /etc/k0s/auth/auth-config.yaml
+    - content: |
+      {{- with .Values.k0s.auth.config }}
+      {{- toYaml . | nindent 8 }}
+      {{- end }}
+      permissions: "0644"
+      path: /etc/k0s/auth/auth-config.yaml
+    {{- range .Values.k0s.files }}
+    - {{ toYaml . | trim | nindent 6 }}
+    {{- end }}
+    {{- else }}
+    files: {{ toYaml .Values.k0s.files | nindent 4 }}
+    {{- end }}
+    {{- else if and (and (.Values.k0s.auth) (.Values.k0s.auth.enabled)) (not .Values.k0s.files) }}
+    files:
+    - content: |
+      {{- with .Values.k0s.auth.config }}
+      {{- toYaml . | nindent 8 }}
       {{- end }}
+      permissions: "0644"
+      path: /etc/k0s/auth/auth-config.yaml
+    {{- else if and (.Values.k0s.files) (not .Values.k0s.auth) }}
+    files: {{ toYaml .Values.k0s.files | nindent 4 }}
+    {{- end }}
```

Comment:
Can this block be simplified? Currently, I find it a little bit excessive. Something like:
```
{{- $auth := .Values.k0s.auth }}
{{- $files := .Values.k0s.files }}
{{- if or (and $auth $auth.enabled) $files }}
files:
{{- if and $auth $auth.enabled }}
  - content: |
{{ toYaml $auth.config | nindent 8 }}
    permissions: "0644"
    path: /etc/k0s/auth/auth-config.yaml
{{- end }}
{{- range $file := $files }}
  - {{ toYaml $file | trim | nindent 4 }}
{{- end }}
{{- end }}
```
Or does it not render properly this way?